CSharpFunctionalExtensions/Result/Dto/ResultDto.cs

```diff
@@ -0,0 +1,26 @@
+﻿#nullable enable
+using System.Text.Json.Serialization;
+
+namespace CSharpFunctionalExtensions
+{
+    public record ResultDto
+    {
+        public ResultDto() { }
+        public ResultDto(string errorMessage) => (ErrorMessage) = (errorMessage);
+
+        public string? ErrorMessage { get; init; }
+        public bool IsSuccess => ErrorMessage is null;
+
+        public static ResultDto Success() => new();
+        public static ResultDto Failure(string errorMessage) => new(errorMessage);
+
+        public static ResultDto<TResult> Success<TResult>(TResult result) => new(result, null);
+        public static ResultDto<TResult> Failure<TResult>(string errorMessage) => new(default!, errorMessage);
+    }
+
+    public record ResultDto<TResult>(TResult Result, string? ErrorMessage)
+    {
+        [JsonIgnore]
+        public bool IsSuccess => ErrorMessage is null;
+    }
+}
```

Comment:
I have to admit I did not used them but it is probably not hard to add them.
Because this code (in is essences) is working well for my full stack between blazor, server and another server that shares interfaces with Result and Result<T> I had much testing for this code.
I don't want to add something I won't be able to really test in some real environment.
I will be happy to help anyone who wants to do `Result<T, TError>` and `UnitResult<TError>`.

Reply:
We can not expect that each caller will use both serialization and deserialization features. The whole idea behind these converters is to allow you to serialize `Results` without any special code. And I think it will be less surprising if users see serialized *all* `Result` properties. 

Please check [how binary serialization is implemented](https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/c66b80ba7093d15e0c549765bec185dbe6fa7805/CSharpFunctionalExtensions/Result/Internal/ResultCommonLogic.cs#L9). Is uses all properties.

If I return a `Result` from my controller, I'd suggest to see all properties which I see in my intellisense, right?

Reply:
So please 
1. Change `(TValue Result, string? ErrorMessage)` to (TValue? Value, string? Error)
2. Add `IsFailure` to both Dtos
3. Remove `[JsonIgnore]` from `IsSuccess`
4. Remove `init;` accessor from `ResultDto.Error`
5. Either remove `init;` acessors from `(TValue Result, string? ErrorMessage)` or convert records to plain classes

Reply:
I agree names should be similar and I will change of the Result to Value. 
DTO should be minimal and compact to save network/stoarge so I am putting it in only the bare minimum.